File: backend/src/main/java/io/github/daegwonkim/backend/enumerate/entity/RentalItemCategory.java
package io.github.daegwonkim.backend.enumerate.entity;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum RentalItemCategory {
    POPULAR("popular"),
    HOUSEHOLD("household"),
    TRAVEL("travel"),
    SPORTS("sports"),
    ELECTRONICS("electronics"),
    FASHION("fashion"),
    CHILDCARE("childcare")
    ;

    private final String fieldName;

    public static RentalItemCategory fromString(String value) {
        if (value == null || value.isBlank()) {
            return POPULAR;
        }

        for (RentalItemCategory category : values()) {
            if (category.fieldName.equalsIgnoreCase(value)) {
                return category;
            }
        }

        return POPULAR;
    }
}

File: backend/domain/src/main/kotlin/io/github/daegwonkim/backend/enumerate/RentalItemCategory.kt
package io.github.daegwonkim.backend.enumerate

enum class RentalItemCategory(
    val label: String
) {
    HOUSEHOLD("가정용품"),
    TRAVEL("여행용품"),
    SPORTS("스포츠/운동"),
    ELECTRONIC("전자기기"),
    FASHION("패션"),
    CHILDCARE("육아/교육")
}